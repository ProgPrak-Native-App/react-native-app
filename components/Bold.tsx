import { Text, TextProps } from "react-native";
import React from "react";

export default function Bold(props: TextProps) {
  return <Text style={[props.style, { fontWeight: "bold" }]} {...props} />;
};
