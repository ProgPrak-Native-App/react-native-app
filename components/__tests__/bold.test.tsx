import React from 'react';
import renderer from 'react-test-renderer';

import Bold from '../Bold';

describe('<Bold />', () => {
  it('Is a Bold component', async () => {
    const tree = renderer.create(<Bold />);
    expect(tree.getInstance()?.type).toBe(<Bold />);
  });
  it('Matches Snapshot', async () => {
    const tree = renderer.create(<Bold />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Text is bold', async () => {
    const tree = renderer.create(<Bold />);
    console.log('tree root', tree.root);
    console.log('tree toTree', tree.toTree());

    const styles = tree.getInstance();
    console.log('styles', styles);
    expect(styles).toBe('bold');
  });

  it('Adding additional property works', async () => {
    // eslint-disable-next-line react-native/no-raw-text
    const tree = renderer.create(<Bold style={{ backgroundColor: 'red' }}>Boldtext</Bold>).toJSON();
    console.log('tree as json', tree);
    expect(tree).toMatchSnapshot();
  });
  afterAll(() => setTimeout(() => process.exit(0), 1000));
});
