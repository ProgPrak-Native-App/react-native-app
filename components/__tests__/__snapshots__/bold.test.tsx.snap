// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bold /> Adding additional property works 1`] = `
<Text
  style={
    Array [
      undefined,
      Object {
        "fontWeight": "bold",
      },
    ]
  }
>
  Boldtext
</Text>
`;

exports[`<Bold /> Matches Snapshot 1`] = `
<Text
  style={
    Array [
      undefined,
      Object {
        "fontWeight": "bold",
      },
    ]
  }
/>
`;
